tests/templates/test_template_inheritance.py

```diff
@@ -0,0 +1,393 @@
+        for typeattr in un_updated_parent_type.typeattrs:
+            if typeattr.id == typeattr_to_update.id:
+                un_updated_parent_type_attr = typeattr
+                break
+
+        #The parent and child type should have the same number of type attrs, but
+        #one of the children's ones will be set to 'X'
+        assert len(original_type.typeattrs) == len(updated_child_type.typeattrs)
+        assert updated_child_type_attr.status == 'X'
+        assert un_updated_parent_type_attr.status == 'A'
+
+    def test_get_updated_typeattr(self, client):
+        """
+            Test the individual 'get_type' functionality by
+            updating a template type attribute in a child, by adding a type attribute
+            to a type in the child template, which has a perent in the parent template,
+            and then setting the status of that to 'X'
+        """
+
+        parent_template_j = client.testutils.create_template()
+
+        child_template_j = client.testutils.create_child_template(parent_template_j.id)
+
+        child_template_received = client.get_template(child_template_j.id)
+
+        #Get a type which inherits from the parent, and update it
+        typeattr_to_update = None
+        original_type = None
+        for templatetype in child_template_received.templatetypes:
+            if templatetype.template_id == parent_template_j.id:
+                original_type = templatetype
+                typeattr_to_update = templatetype.typeattrs[0]
+                break
+
+        #Create a child of a parent *typeattr*, so we can update it.
+        #When the child template is requested, this typeattr will contain the combined
+        #data from itself and from its parent.
+        child_typeattr = client.add_child_typeattr(typeattr_to_update.id, child_template_j.id)
+
+        child_typeattr.status = 'X'
+
+        client.update_typeattr(child_typeattr)
+
+        updated_child_type_attr = client.get_typeattr(child_typeattr.id)
+        un_updated_parent_type_attr = client.get_typeattr(typeattr_to_update.id)
+
+        assert updated_child_type_attr.status == 'X'
+        assert un_updated_parent_type_attr.status == 'A'
+
+
+    def test_create_network_with_child_template(self, client):
+        """
+            This tests that a network is correctly linked with its template upon
+            creation. The issues is:
+            Upon creating using a template, a network is linked to a template with
+            the template's 'network' template type. In an inherited template, this
+            template type may belong to the parent, and so there is no direct means
+            of linking a network to the correct child template, as the type ID of the 'network'
+            template id will come from the parent. To mitigate this, a 'child template id' column
+            exists on the tResourceType table which allows us to explicitly say which template was
+            used to create the network..
```

Comment:
What are the implications of having the `network` template type be the `child_template_id` directly where a network is created with a child template?

Reply:
When creating a network using a child template, it's possible that there is no TemplateType entry for the network within the child template i.e. there's no entry in the DB for that template type. This means that when you request the template, the template type which is returned is that of the parent, and the type_id points to the template type  entry in the parent template.

If we store only this type_id, then we have lost the connection to the child template, as Hydra will only store the link to the type which is in the parent, and therefore hydra will think that the network was created using the parent template.

Reply:
I've put this response text into the test docstring